### PR TITLE
Fix ACL copying with Blob

### PR DESCRIPTION
**Pull request by adreed-msft**
Somewhere along the way, when we moved to generating clients in `cmd`, we opted to generate a dfs and blob client when working with BlobFS, but we didn't do so for working with Blob.

This
1) Creates a DFS and Blob client as long as we're working with either resource, because they're the same resource
2) Enables some feature-parity with sync where specifying include-directory-stubs is used whenever persisting permissions on Blob, because we know that we're working with HNS and will encounter folders
3) Allows our testing suite to *actually* target BlobFS literally, rather than pretending.

### CODE REVIEW

**Review by gapra-msft — APPROVED** [review]

**Review by tasherif-msft — DISMISSED** [review]

**Review by nakulkar-msft — APPROVED** [review]

**Review by gapra-msft — DISMISSED** [review]

File: e2etest/zt_basic_cli_ps_auth_test.go [hunk 1]
```diff
@@ -1,176 +1,176 @@
...
-		defaultSize: "1K",
-		shouldTransfer: []interface{}{
-			"wantedfile",
-			folder("sub/subsub"),
-			"sub/subsub/filea",
-			"sub/subsub/filec",
-		},
-	}, EAccountType.Standard(), EAccountType.Standard(), "")
-}
+//// Purpose: Tests AZCLI and powershell auth tests
```
Comment (siminsavani-msft) [inline e2etest/zt_basic_cli_ps_auth_test.go:32]: Why are these tests removed? /gen

File: e2etest/zt_basic_copy_sync_remove_test.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ import (
 
 // ================================  Copy And Sync: Upload, Download, and S2S  =========================================
 func TestBasic_CopyUploadSingleBlob(t *testing.T) {
-	RunScenarios(t, eOperation.CopyAndSync(), eTestFromTo.AllUploads(), eValidate.AutoPlusContent(), anonymousAuthOnly, allCredentialTypes, params{
+	RunScenarios(t, eOperation.Copy(), eTestFromTo.Other(common.EFromTo.LocalBlobFS()), eValidate.AutoPlusContent(), anonymousAuthOnly, anonymousAuthOnly, params{
```
Comment (siminsavani-msft) [inline e2etest/zt_basic_copy_sync_remove_test.go:44]: Would it be possible to add LocalBlobFS to AllUploads? What about testing LocalFile?

**Review by siminsavani-msft — APPROVED** [review]
Had some questions about the e2e tests modifications, otherwise, LGTM!

File: cmd/syncEnumerator.go [hunk 3]
```diff
@@ -55,12 +55,14 @@ func (cca *cookedSyncCmdArgs) initEnumerator(ctx context.Context) (enumerator *s
 		}
 	}
 
+	includeDirStubs := cca.fromTo.From().SupportsHnsACLs() && cca.fromTo.To().SupportsHnsACLs() && cca.preservePermissions.IsTruthy()
```
Comment (siminsavani-msft) [inline cmd/syncEnumerator.go:58]: Would it make sense to replace the isHNSToHNS option cookedSyncCmdArgs to have includeDirStubs?

File: e2etest/zt_basic_copy_sync_remove_test.go (hunk 2 above)
Comment (gapra-msft) [inline e2etest/zt_basic_copy_sync_remove_test.go:44]: was this changed just for a temporary test?

File: e2etest/zt_basic_copy_sync_remove_test.go [hunk 4]
```diff
@@ -419,8 +419,19 @@ func TestBasic_CopyRemoveFolderHNS(t *testing.T) {
 }
 
 func TestBasic_CopyRemoveContainer(t *testing.T) {
+	bfsRemove := TestFromTo{
```
Comment (gapra-msft) [inline e2etest/zt_basic_copy_sync_remove_test.go:422]:
```suggestion
	allButBfsRemove := TestFromTo{
```

File: e2etest/zt_basic_cli_ps_auth_test.go (hunk 1 above)
Comment (adreed-msft) [reply e2etest/zt_basic_cli_ps_auth_test.go]: Accidental, realized later. I was running tests locally, these tests panic if you don't have the right tools.

File: e2etest/zt_basic_copy_sync_remove_test.go (hunk 2 above)
Comment (adreed-msft) [reply e2etest/zt_basic_copy_sync_remove_test.go:44]: Coincidentally, I think this was another accident. I believe I'd just reduced the test conditions to test exactly what I was working on.

Comment (adreed-msft) [reply e2etest/zt_basic_copy_sync_remove_test.go:44]: LocalBlobFS was present

File: cmd/syncEnumerator.go (hunk 3 above)
Comment (adreed-msft) [reply cmd/syncEnumerator.go:58]: isHNStoHNS hasn't existed for an extended period of time.

File: e2etest/zt_enumeration_filter_test.go [hunk 5]
```diff
@@ -162,21 +162,6 @@ func TestFilter_RemoveFolder(t *testing.T) {
 	}, EAccountType.Standard(), EAccountType.Standard(), "")
 }
 
-func TestFilter_RemoveContainer(t *testing.T) {
```
Comment (gapra-msft) [inline e2etest/zt_enumeration_filter_test.go:165]: why is this test removed? unrelated but this seems like its not testing what the test title claims

File: e2etest/zt_preserve_properties_test.go [hunk 6]
```diff
@@ -44,7 +44,7 @@ func TestProperties_NameValueMetadataIsPreservedS2S(t *testing.T) {
 
 func TestProperties_NameValueMetadataCanBeUploaded(t *testing.T) {
 	expectedMap := map[string]*string{"foo": to.Ptr("abc"), "bar": to.Ptr("def"), "baz": to.Ptr("state=a;b")}
-	RunScenarios(t, eOperation.Copy(), eTestFromTo.AllUploads(), eValidate.Auto(), anonymousAuthOnly, anonymousAuthOnly, params{
+	RunScenarios(t, eOperation.Copy(), eTestFromTo.Other(common.EFromTo.LocalBlobFS()), eValidate.Auto(), anonymousAuthOnly, anonymousAuthOnly, params{
```
Comment (gapra-msft) [inline e2etest/zt_preserve_properties_test.go:47]: double check this

File: e2etest/zt_basic_copy_sync_remove_test.go [hunk 7]
```diff
@@ -869,7 +879,7 @@ func TestBasic_OverwriteHNSDirWithChildren(t *testing.T) {
 	RunScenarios(
 		t,
 		eOperation.Copy(),
-		eTestFromTo.Other(common.EFromTo.LocalBlobFS()),
+		eTestFromTo.Other(common.EFromTo.BlobFSBlobFS()),
```
Comment (siminsavani-msft) [inline e2etest/zt_basic_copy_sync_remove_test.go:882]: Why is LocalBlobFS removed?

File: e2etest/zt_preserve_properties_test.go [hunk 8]
```diff
@@ -57,7 +57,7 @@ func TestProperties_NameValueMetadataCanBeUploaded(t *testing.T) {
 }
 
 func TestProperties_HNSACLs(t *testing.T) {
-	RunScenarios(t, eOperation.CopyAndSync(), eTestFromTo.Other(common.EFromTo.BlobBlob()), eValidate.Auto(), anonymousAuthOnly, anonymousAuthOnly, params{
+	RunScenarios(t, eOperation.Copy(), eTestFromTo.Other(common.EFromTo.BlobBlob(), common.EFromTo.BlobFSBlobFS(), common.EFromTo.BlobBlobFS(), common.EFromTo.BlobFSBlob()), eValidate.Auto(), anonymousAuthOnly, anonymousAuthOnly, params{
```
Comment (siminsavani-msft) [inline e2etest/zt_preserve_properties_test.go:60]: Should we add a separate test for BlobBlob for sync?

File: e2etest/zt_enumeration_filter_test.go (hunk 5 above)
Comment (adreed-msft) [reply e2etest/zt_enumeration_filter_test.go:165]: This was a duplicate of another test with a different name that didn't actually test the relevant functionality that was attempted to be tested. I could rewrite this test and add it back in if wanted, but I believe it relies upon account-wide testing functionality, which, makes it probably not something to pursue in this framework.

Comment (adreed-msft) [reply e2etest/zt_enumeration_filter_test.go:165]: (Could probably test it in the new E2E framework though!)

Comment (adreed-msft) [reply e2etest/zt_enumeration_filter_test.go:165]: Actually. I'm not even sure what this is supposed to test. I guess using a wildcard in container name? idk. The name doesn't provide much information, and the test, clearly not testing what it's trying to test, does not make much sense.

**Review by gapra-msft — APPROVED** [review]

**Pull request merged**